src/main/scala/com.socrata.geoexport/encoders/intermediates/KML.scala

```diff
@@ -0,0 +1,266 @@
+}
+
+class DoubleRep(soqlName: String) extends KMLRep[SoQLDouble](soqlName) with SimpleDatum {
+  def toAttrValues(soql: SoQLDouble): Seq[Any] = asSimpleData(soql.value.toString)
+}
+
+class ArrayRep(soqlName: String) extends KMLRep[SoQLArray](soqlName) with SimpleDatum {
+  def toAttrValues(soql: SoQLArray): Seq[Any] = {
+    asSimpleData(soql.value.toArray.map {
+      case JString(s) => s
+      case other: Any => other
+    }.mkString(", "))
+  }
+}
+
+/**
+  Complex intermediate mappings.
+
+  If we have
+  ```
+    {"i": "love", {"xml": "jk"}}
+  ```
+
+  That would get mapped and flattened to
+  ```
+    <SimpleData name="i">love</SimpleData>
+    <SimpleData name="i.xml">jk</SimpleData>
+  ```
+*/
+class JSONRep(soqlName: String) extends KMLRep[SoQLJson](soqlName) with ComplexDatum {
+  def toAttrValues(soql: SoQLJson): Seq[Any] = fromJValue(soqlName, soql.value)
+}
+
+class ObjectRep(soqlName: String) extends KMLRep[SoQLObject](soqlName) with ComplexDatum {
+  def toAttrValues(soql: SoQLObject): Seq[Any] = fromJValue(soqlName, soql.value)
+}
+
+/**
+  Maps all the SoQLColumns to intermediate columns
+*/
+object KMLRepMapper extends RepMapper {
+  def forPoint(name: String): PointRep =                          new PointRep(name)
+  def forMultiPoint(name: String): MultiPointRep =                new MultiPointRep(name)
+  def forLine(name: String): LineRep =                            new LineRep(name)
+  def forMultiLine(name: String): MultiLineRep =                  new MultiLineRep(name)
+  def forPolygon(name: String): PolygonRep =                      new PolygonRep(name)
+  def forMultiPolygon(name: String): MultiPolygonRep =            new MultiPolygonRep(name)
+  def forDate(name: String): DateRep =                            new DateRep(name)
+  def forTime(name: String): TimeRep =                            new TimeRep(name)
+  def forFloatingTimestamp(name: String): FloatingTimestampRep =  new FloatingTimestampRep(name)
+  def forFixedTimestamp(name: String): FixedTimestampRep =        new FixedTimestampRep(name)
+  def forNumber(name: String): NumberRep =                        new NumberRep(name)
+  def forText(name: String): TextRep =                            new TextRep(name)
+  def forMoney(name: String): MoneyRep =                          new MoneyRep(name)
+  def forBoolean(name: String): BooleanRep =                      new BooleanRep(name)
+  def forVersion(name: String): VersionRep =                      new VersionRep(name)
+  def forID(name: String): IDRep =                                new IDRep(name)
+  def forArray(name: String): ArrayRep =                          new ArrayRep(name)
+  def forDouble(name: String): DoubleRep =                        new DoubleRep(name)
+  def forJson(name: String): JSONRep =                            new JSONRep(name)
+  def forObject(name: String): ObjectRep =                        new ObjectRep(name)
+  // scalastyle:off
```

Comment:
Maybe silence the specific warnings, instead of all of them?  It makes it more apparent what's going on.